be/src/storage/lake/pk_tablet_writer.cpp

```diff
@@ -111,6 +135,11 @@ Status HorizontalPkTabletWriter::flush_segment_writer(SegmentPB* segment) {
         }
         _seg_writer.reset();
     }
+    if (_pk_sst_writer != nullptr) {
```

Comment:
is it possible to have a dummy sst writer, the _pk_sst_writer will be always created. If not enabled, it will be  created as a dummy writer? kind of eliminating the check nullptr of the writer, make the code less risky of nullptr crash.